service/src/main/java/uk/nhs/adaptors/gpc/consumer/LoggingGatewayFilterFactory.java

```diff
@@ -0,0 +1,104 @@
+package uk.nhs.adaptors.gpc.consumer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.reactivestreams.Publisher;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.GatewayFilterChain;
+import org.springframework.cloud.gateway.filter.OrderedGatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.http.server.reactive.ServerHttpResponseDecorator;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+@Component
+@Slf4j
+public class LoggingGatewayFilterFactory extends AbstractGatewayFilterFactory<LoggingGatewayFilterFactory.Config> {
+    private static final List<String> LOGGABLE_HEADER_KEYS = List.of("Ssp-From", "Ssp-To");
+    private static final String LOG_TEMPLATE = "Gateway filter log: %s %s URL: %s";
+    private static final String HEADERS_PREFIX = "Headers: { ";
+    private static final String HEADERS_SUFFIX = "}";
+    private static final String COLON = ": ";
```

Comment:
Can we use `key=value` format for logging? This would consistent with what is done already in other adaptors.